snakebite/client.py

```diff
@@ -1425,10 +1476,16 @@ def __handle_socket_error(self, exception):
         log.debug("Request failed with %s" % exception)
         if exception.errno in (errno.ECONNREFUSED, errno.EHOSTUNREACH):
             # if NN is down or machine is not available, pass it:
-            pass
+            if self.__should_retry():
```

Comment:
You are right, I ignored that point. The Java client behavior is actually to failover directly on the following exceptions: ConnectException, NoRouteToHostException, UnknownHostException, StandbyException and ConnectTimeoutException. It also has a second limit for max failovers which is 15 by default.

Considering that the clients might otherwise freeze when the configured first NN is down. I'll try to implement that behavior.